### ISSUE TITLE
conductor: remove unnecessary commit verification 

### ISSUE THREAD

**Issue by noot** [astriaorg/astria#401]
as we're now using a trusted sequencer endpoint in conductor, we don't need to validate commits anymore, specifically on data received from DA. we can simply use some `get_header_by_height` endpoint and confirm that the block header (and thus hash) returned (which we know is final) matches the block hash on DA.

as such, we can remove `last_commit` from `SequencerBlockData`.

**Comment by SuperFluffy** [comment]
Is this relevant for #395? Should we adjust our protos accordingly and not send them?

**Comment by noot** [comment]
I thought about this some more and I'm concerned about the verification as a whole - if we decide conductor fully trusts the sequencer endpoint, then not only should we remove `last_commit`, but we can remove all the block verification entirely (including things like verifying `action_tree_root` is inside `data_hash`). Since the sequencer node can just send us any data it wishes, and we trust it, there's no need for any validation at all (and if it wants to forge rollup transactions, it can make a different `data_hash` etc). this seems concerning to me, I will discuss it with @joroshiba

**Comment by noot** [comment]
I'm leaning towards still using a sequencer RPC endpoint but embedding the tendermint-rs light client inside conductor so that conductor can use semi-trusted remote nodes; ie. it doesn't need to fully trust the data it receives from the node, and if it receives invalid data it will be notified via some error and need to change to a different node. anyways needs discussion!

**Comment by joroshiba** [comment]
I agree having the semi trusted methodology would still be better, it would be ideal for conductor to be able to utilize a remote node while receiving information about validator sets via p2p in order to validate.

Related to this is the question of if we should provide a mechanism by which to grab only a specific rollups data from the sequencer. This would reduce over wire data transfer from a node to a conductor, with the requirement to validate all data in the same way as from celestia. Theoretically could be same data shape as posted to DA.
